File: multi_language_code_generator/tests/python/test_order.py
import unittest
import sys
import os

# Implimentasiya qovlugunu yola ?lav? edirik
sys.path.append(os.path.abspath(os.path.join(os.path.dirname(__file__), '../../implementations/python')))
from order_processor import process_order

class TestOrderProcessor(unittest.TestCase):
    def test_equivalence(self):
        result = process_order("Laptop", 1200, 2)
        self.assertEqual(result["total"], 2400)
        self.assertEqual(result["status"], "Processed")

if __name__ == '__main__':
    unittest.main()

File: multi_language_code_generator/tests/javascript/test_order.js
const assert = require('assert');
const { processOrder } = require('../../implementations/javascript/order_processor');

// Qeyd: JS faylinda 'module.exports = { processOrder };' ?lav? olunmalidir
try {
    const result = processOrder("Laptop", 1200, 2);
    assert.strictEqual(result.total, 2400);
    assert.strictEqual(result.status, "Processed");
    console.log("JS Test Passed!");
} catch (err) {
    console.error("JS Test Failed:", err);
    process.exit(1);
}
